tls/s2n_ktls_io.c

```diff
@@ -146,6 +146,13 @@ S2N_RESULT s2n_ktls_get_control_data(struct msghdr *msg, int cmsg_type, uint8_t
     RESULT_ENSURE_REF(msg);
     RESULT_ENSURE_REF(record_type);
 
+    /* https://man7.org/linux/man-pages/man3/recvmsg.3p.html
+     * MSG_CTRUNC  Control data was truncated.
+     */
+    if (msg->msg_flags & MSG_CTRUNC) {
```

Comment:
Do we need to check any other msg_flags? MSG_TRUNC looks interesting.

Reply:
I think `MSG_TRUNC` would be the only other interesting error flag we would care about but it only applies to UDP so we are ok not checking for this.

> indicates that the trailing portion of a `datagram` was discarded because the datagram was larger than the buffer supplied. 

---
Since you asked :D openssl's FreeBSD impl does a really weird check using EOR and CTRUNC flag. Once we support kTLS on FreeBSD we will possibly need to check at that flag. https://github.com/aws/s2n-tls/issues/4153